### PR TITLE
Make it possible to configure the MongoDB connection

### PR DESCRIPTION
**Pull request by wjzijderveld**

### CODE REVIEW

File: src/Broadway/Bundle/BroadwayBundle/DependencyInjection/BroadwayExtension.php [hunk 1]
```diff
@@ -70,7 +70,27 @@ private function loadSagaStateRepository(array $config, ContainerBuilder $contai
...
+                if (isset($config['mongodb']['connection'])) {
+
+                    if (isset($config['mongodb']['connection']['database'])) {
+                        $database = $config['mongodb']['connection']['database'];
+                    }
+
+                    $mongoConnection = $container->getDefinition('broadway.saga.state.mongodb_connection');
+
+                    if (isset($config['mongodb']['connection']['dsn'])) {
+                        $mongoConnection->replaceArgument(0, $config['mongodb']['connection']['dsn']);
```
Comment (marcojanssen) [inline src/Broadway/Bundle/BroadwayBundle/DependencyInjection/BroadwayExtension.php]: replace argument 0? can't specifiy it ?

```
setDsn() ?
```

Comment (wjzijderveld) [inline src/Broadway/Bundle/BroadwayBundle/DependencyInjection/BroadwayExtension.php]: Nope, sadly not. Tried just collecting and setting the arguments, but that didn't work well, as both are optional...

File: test/Broadway/EventSourcing/AbstractEventSourcingRepositoryTest.php [hunk 2]
```diff
@@ -164,7 +164,7 @@ public function it_publishes_decorated_events()
         $repository = new EventSourcingRepository(
             $this->eventStore,
             $this->eventBus,
-            '\Broadway\EventSourcing\TestEventSourcedAggregate',
+            get_class($this->createAggregate()),
```
Comment (asm89) [inline test/Broadway/EventSourcing/AbstractEventSourcingRepositoryTest.php]: This change doesn't seem related. Can you make it in a separate PR?

**Comment by asm89** [comment]
A comment, otherwise +1.

File: test/Broadway/EventSourcing/AbstractEventSourcingRepositoryTest.php (hunk 2 above)
Comment (wjzijderveld) [inline test/Broadway/EventSourcing/AbstractEventSourcingRepositoryTest.php]: I can, but my tests failed because of this... Not sure why this would
suddenly break though. Will look at it again..

**Comment by othillo** [comment]
:+1:
